public/app/features/dashboard/time_srv.ts

```diff
@@ -158,15 +157,23 @@ export class TimeSrv {
   }
 
   setTime(time, fromRouteUpdate?) {
+    const oldRange = this.timeRange();
+
     _.extend(this.time, time);
 
-    // disable refresh if zoom in or zoom out
+    // Disable refresh when 'to' is a fixed time
     if (moment.isMoment(time.to)) {
-      this.oldRefresh = this.dashboard.refresh || this.oldRefresh;
       this.setAutoRefresh(false);
-    } else if (this.oldRefresh && this.oldRefresh !== this.dashboard.refresh) {
-      this.setAutoRefresh(this.oldRefresh);
-      this.oldRefresh = null;
+    }
+
+    // Disable refresh when zooming out
+    if (this.dashboard.refresh) {
+      const newRange = this.timeRange();
+      const newSpan = newRange.to.valueOf() - newRange.from.valueOf();
+      const oldSpan = oldRange.to.valueOf() - oldRange.from.valueOf();
+      if (newSpan > oldSpan * 1.2) {
```

Comment:
I am afraid that his is way to low and will be annoying. Think it should only trigger when increasing time range by say 3x.  it will trigger if you go from 1h, to 6h, or 1d, to 7days, or 1month, to 12months, think 1.2x is way to low as it will disable auto refresh always (basically) when you just do small changes